agent/acs/handler/attach_eni_handler.go

```diff
@@ -118,14 +128,39 @@ func (attachENIHandler *attachENIHandler) handleSingleMessage(message *ecsacs.At
 // addENIAttachmentToState adds the eni info to the state
 func (handler *attachENIHandler) addENIAttachmentToState(message *ecsacs.AttachTaskNetworkInterfacesMessage) {
 	attachmentArn := aws.StringValue(message.ElasticNetworkInterfaces[0].AttachmentArn)
+	eniMac := aws.StringValue(message.ElasticNetworkInterfaces[0].MacAddress)
+	// Stop tracking the eni attachment after timeout
+	eniAckTimeout := time.Duration(aws.Int64Value(message.WaitTimeoutMs)) * time.Millisecond
+	eniAckTimeoutHandler := ackTimeoutHandler{mac: eniMac, state: handler.state}
+	ackTimeoutTimer := time.AfterFunc(eniAckTimeout, eniAckTimeoutHandler.handle)
+
 	seelog.Info("Adding eni info to state, eni: %s", attachmentArn)
-	handler.state.AddENIAttachment(
-		&api.ENIAttachment{
-			TaskArn:          aws.StringValue(message.TaskArn),
-			AttachmentArn:    attachmentArn,
-			AttachStatusSent: false,
-			MacAddress:       aws.StringValue(message.ElasticNetworkInterfaces[0].MacAddress),
-		})
+	eniAttachment := &api.ENIAttachment{
+		TaskArn:          aws.StringValue(message.TaskArn),
+		AttachmentArn:    attachmentArn,
+		AttachStatusSent: false,
+		MacAddress:       eniMac,
+		AckTimer:         ackTimeoutTimer,
+	}
+	handler.state.AddENIAttachment(eniAttachment)
+}
+
+// ackTimeoutHandler remove ENI attachment from agent state after the ENI ack timeout
+type ackTimeoutHandler struct {
+	mac   string
+	state dockerstate.TaskEngineState
+}
+
+func (handler *ackTimeoutHandler) handle() {
+	eniAttachment, ok := handler.state.ENIByMac(handler.mac)
+	if !ok {
+		seelog.Warnf("ENI ack timeout handler: ENI attachment was not managed by agent with MAC address: %s", handler.mac)
```

Comment:
Can we make this less verbose? `"Timed out waiting for ENI ack; ignoring unmanaged ENI attachment with MAC address: %s"`